instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/item/ChunkOrientedTaskletInstrumentation.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.item;
+
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.SpringBatchInstrumentationConfig.isItemLevelTracingEnabled;
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.item.ItemTracer.tracer;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.batch.core.scope.context.ChunkContext;
+
+public class ChunkOrientedTaskletInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    return named("org.springframework.batch.core.step.item.ChunkOrientedTasklet");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isPublic()
+            .and(named("execute"))
+            .and(takesArguments(2))
+            .and(takesArgument(0, named("org.springframework.batch.core.StepContribution")))
+            .and(
+                takesArgument(
+                    1, named("org.springframework.batch.core.scope.context.ChunkContext"))),
+        this.getClass().getName() + "$ExecuteAdvice");
+  }
+
+  public static class ExecuteAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter(@Advice.Argument(1) ChunkContext chunkContext) {
+      if (isItemLevelTracingEnabled()) {
```

Comment:
So we always instrument classes, even if we actually end up doing nothing? Not sure if this is a good idea...

Reply:
I think that since this is a static final boolean constant JIT will remove this code and it won't matter at all in the end. But I agree that the previous version looked better - muzzle had problems accepting it though (couldn't see the instrumentations that are disabled by default) so I decided checking the flag in advice classes is still better than `additionalHelperClassNames()`